### PR TITLE
implement audit logs in caesar-common

### PR DESCRIPTION
**Pull request by vaibhavac**

### CODE REVIEW

File: common/rest/middleware/changeLog.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def get_current_request():
+    return _request_ctx.get()
+
+def add_to_request_context(key, value):
+    context = _request_ctx.get()
+    context[key] = value
+    _request_ctx.set(context)
+
+def get_ip_address():
+    try:
```
Comment (suryadana) [inline common/rest/middleware/changeLog.py]: This method should get ip public on the server not from x-real or x-forwarded-for this is not true logic.

File: common/rest/signals/receivers.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    return field.related_model and field.related_model._meta.model_name == 'agent'
+
+@receiver(pre_save)
+def capture_old_values(sender, instance, **kwargs):
+    pass
+    # if skip_agent_model(sender):
+    #     return
+
+    # if instance.pk:
+    #     try:
```
Comment (suryadana) [inline common/rest/signals/receivers.py]: remove the code already commented

**Review by suryadana — CHANGES_REQUESTED** [review]

File: common/rest/middleware/changeLog.py (hunk 1 above)
Comment (vaibhavac) [reply common/rest/middleware/changeLog.py]: Fixed. The method now correctly retrieves the public IP from the server.

File: common/rest/signals/receivers.py (hunk 2 above)
Comment (vaibhavac) [reply common/rest/signals/receivers.py]: I've removed the commented code as requested.

**Review by suryadana — APPROVED** [review]

**Pull request merged**